src/Microsoft.Framework.BufferedHtmlContent.Sources/BufferedHtmlContent.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNet.Html.Abstractions;
+using Microsoft.Framework.WebEncoders;
+
+namespace Microsoft.Framework.Internal
+{
+    /// <summary>
+    /// Enumerable object collection which knows how to write itself.
+    /// </summary>
+    internal class BufferedHtmlContent : IHtmlContent, IEnumerable<object>
```

Comment:
The above `Append` method is assuming that `TagHelperContent.Append` can be called with `string` and `TagHelperContent` only. If we change `TagBuilder` to implement `IHtmlContent` as pointed out <a href="https://github.com/aspnet/Mvc/pull/2899/files#diff-3d2f28336a81217f32fda542472d8441L252">here</a>, that would be `Append`ed to `TagHelperContent` as well. We would end up converting `TagBuilder`s (and other `IHtmlContent`s) to strings even if `IsEmpty`/`IsWhitespace` is never called. So I think this is a bad idea.

The enumeration is not used anywhere apart from the *check* methods in `DefaultTagHelperContent`. I'll remove `BufferedHtmlContent` from being an `IEnumerable`. Instead I'll add constructor like:
```C#
public BufferedHtmlContent(List<object> store)
```
This would initialize the `BufferedHtmlContent`'s backing store with the one passed in. With this, I can solve the enumeration problem and also conversion to strings happen only when necessary. What do you think?